pymeasure/instruments/newport/smc100.py

```diff
@@ -0,0 +1,709 @@
+                except IndexError:
+                    pass
+                message += f"* {k}: {v.fget(self)}{unit}\n"
+        return message[:-1]
+
+    #Implementation or extension of Instrument parent class methods
+    def check_errors(self):
+        """Check for command error except when broadcasting."""
+        if self.address != "0":
+            self.check_command_error()
+
+    def ask(self, command):
+        return super().ask(self.address + command)
+
+    def shutdown(self):
+        """Stop any motion and put the controller in DISABLE state."""
+        self.stop()
+        self.wait_ready()
+        self.disable = True
+        super().shutdown()
+
+    def values(self, command, **kwargs):
+        return super().values(self.address + command, **kwargs)
+
+    def write(self, command):
+        super().write(self.address + command)
+
+    #Other methods
+    def check_command_error(self):
+        """Check for and raise any command error."""
+        error_code = self.error_command
+        try:
+            assert error_code == '@'
+        except AssertionError:
+            raise(CommandError(self.address, error_code))
+
+    def check_motion_error(self):
+        """Check for and raise any motion error.
+        
+        It implements this by verifying that the controller doesn't
+        unexpectedly end up in DISABLE or NOT REFERENCED states, which
+        indicates that an error happened during homing or moving. Using
+        this command may hide positioner errors.
+        """
+        state = self.state
+        try:
+            assert not state.disable and not state.not_referenced
+        except AssertionError:
+            raise(MotionError(self.address, state))
+
+    def check_positioner_error(self):
+        """Check for and raise any positioner error."""
+        errors = self.error_positioner
+        try:
+            assert not any(errors)
+        except AssertionError:
+            raise(PositionerError(self.address, errors))
+    
+    def get_motion_time(self, distance):
+        """Get an evaluation of the time needed to complete a
+        relative move.
+
+        Argument: distance of the move in predefined unit.
+        """
+        return self.values("PT" + str(abs(distance)))[0]
+
+    def reset(self):
+        """Reset the controller."""
+        self.write("RS")
+        sleep(1)
+        #Sequence necessary to avoid error on next command
+        try:
+            self.check_errors()
+        except VisaIOError:
+            pass
+
+    def search_home(self, type=None, velocity=None, time_out=None):
+        """Execute home search command.
+        
+        Keyword arguments:
+        * type: type of home search
+        * velocity: velocity used for home search in u/s
+        * time_out: maximum time allowed to complete home search in s
+        """
+        if type is not None:
+            self.home_search_type = type
+            self.check_errors()
```

Comment:
Are these `check_errors` calls for debug only?

Reply:
Please see my other answers on the same subject.

Reply:
Ok